lib/model/index.js

```diff
@@ -1,358 +1,373 @@
+        tableName,
+        pkName = 'id',
+        enumMapping,
+        autoSetTimeFields = globalAutoSetTimeFields // v0.1.8 for whole model to use
+      } = args
+      /**
+       * @member
+       * @description the name of table which you want to operate
+       */
+      this.tableName = tableName
+      /**
+       * @member
+       * @description the name of primary key
+       */
+      this.pkName = pkName
+      /**
+       * @member
+       * @description the mapping for enum
+       */
+      this.enumMapping = enumMapping
+      /**
+       * @member
+       * @description array of column names that should be automatically updated with a current timestamp
+       */
+      this.autoSetTimeFields = autoSetTimeFields
+      /**
+       * @member
+       * @description the database operator
+       */
+      this.dataAccess = new DataAccess(connection)
+    }
 
-		/**
-		 * @method
-		 * @param {function} callback Function to be run before comitting the database operation
-		 * @description query without conditions for one table
-		 */
-		async findAll(callback) {
-			try {
-				const res = await this.dataAccess.SingleQueryExecutor({
-					tableName: this.tableName,
-					callback
-				})
-				return res ? this.decodeToEnum(res) : []
-			} catch (e) {
-				throw e
-			}
-		}
+    /**
+     * @method
+     * @param {function} callback Function to be run before comitting the database operation
+     * @description query without conditions for one table
+     */
+    async findAll(callback) {
+      try {
+        const res = await this.dataAccess.SingleQueryExecutor({
+          tableName: this.tableName,
+          callback
+        })
+        return res ? this.decodeToEnum(res) : []
+      } catch (e) {
+        throw e
+      }
+    }
 
-		/**
-		 * @method
-		 * @description query by primary key for one table
-		 * @param {string|number} pk primary key value
-		 * @param {string} selectFields which columns you want to query, default '*'
-		 * @param {function} callback Function to be run before comitting the database operation
-		 */
-		async findByPk(pk, selectFields = '*', callback) {
-			try {
-				const res = await this.dataAccess.SingleQueryExecutor({
-					tableName: this.tableName,
-					whereClause: `"${this.pkName}" = '${pk}'`,
-					selectFields,
-					callback,
-					returnSingleRecord: true
-				})
-				return res ? this.decodeToEnum(res) : {}
-			} catch (e) {
-				throw e
-			}
-		}
+    /**
+     * @method
+     * @description query by primary key for one table
+     * @param {string|number} pk primary key value
+     * @param {string} selectFields which columns you want to query, default '*'
+     * @param {function} callback Function to be run before comitting the database operation
+     */
+    async findByPk(pk, selectFields = '*', callback) {
+      try {
+        let whereClause = '1 = 1'
+        const pks = this.pkName.split(',')
+        if (pks.length > 1) {
+          if (Object.keys(pk).length > 0) {
```

Comment:
I just tried `Object.keys('3')` in node and got `['0']`. I don't think that would be the inteded behavior, correct?

Reply:
I forgot to judge the type. I fixed it.